frontend/src/chrome/IconLink.tsx

```diff
@@ -1,21 +1,42 @@
 import React from 'react'
 import ExternalLink from './ExternalLink'
-import Icon from './Icon'
+import Icon, { IconSize } from './Icon'
 
 interface IconLinkProps {
   icon: string
-  link: string
+  link?: string
+  onClick?: () => void
+  size?: IconSize
 }
 
 const IconLink: React.FC<IconLinkProps> = ({
   icon,
-  link
-}) => (
-  <div className='ml-2'>
-    <ExternalLink to={link} className='text-qrinavy hover:text-qrinavy-600'>
-      <Icon icon={icon} size='sm'/>
-    </ExternalLink>
-  </div>
-)
+  link,
+  onClick,
+  size='sm'
+}) => {
+  const iconComponent = <Icon icon={icon} size={size} />
+  const classNames = 'text-qrinavy hover:text-qripink hover:cursor-pointer'
+
+  if (link) {
+    return (
+      <div className='ml-2'>
+        <ExternalLink to={link} className={classNames}>
+          {iconComponent}
+        </ExternalLink>
+      </div>
+    )
+  }
+
+  if (onClick) {
```

Comment:
no need for the `if` condition. when `onClick` is undefined it shouldn't affect the div at all. This also avoids returning `undefined` from this component if neither `link` or `onClick` are provided